app/models/reservation.rb

```diff
@@ -0,0 +1,22 @@
+class Reservation < ActiveRecord::Base
+  belongs_to :request
+  has_one :donor
+
+  state_machine :initial => :unreserved do
```

Comment:
on an unrelated note, reservations should probably start in the `:reserved` state, since we won't be creating a new reservation except to log a reserved product.